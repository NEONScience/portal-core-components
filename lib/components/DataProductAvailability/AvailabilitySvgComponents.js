"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.SvgDefs = exports.JsxCell = exports.CELL_ATTRS = void 0;
var _react = _interopRequireDefault(require("react"));
var _propTypes = _interopRequireDefault(require("prop-types"));
var _AvailabilityUtils = require("./AvailabilityUtils");
var _Theme = _interopRequireWildcard(require("../Theme/Theme"));
const _excluded = ["nudge"];
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function (e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
function _objectWithoutPropertiesLoose(r, e) { if (null == r) return {}; var t = {}; for (var n in r) if ({}.hasOwnProperty.call(r, n)) { if (e.includes(n)) continue; t[n] = r[n]; } return t; }
function _extends() { return _extends = Object.assign ? Object.assign.bind() : function (n) { for (var e = 1; e < arguments.length; e++) { var t = arguments[e]; for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]); } return n; }, _extends.apply(null, arguments); } /* eslint-disable import/prefer-default-export, max-len */
/**
   SVG_DEFS
*/
const DiagLinesPattern = props => {
  const {
    id,
    color,
    secondaryColor
  } = props;
  const width = _AvailabilityUtils.SVG.CELL_WIDTH;
  const height = _AvailabilityUtils.SVG.CELL_HEIGHT / 4;
  return /*#__PURE__*/_react.default.createElement("pattern", {
    id: id,
    width: width,
    height: height,
    patternUnits: "userSpaceOnUse",
    patternTransform: "rotate(45)"
  }, /*#__PURE__*/_react.default.createElement("rect", {
    x: 0,
    y: 0,
    width: width,
    height: height / 2 - 0.25,
    fill: color
  }), /*#__PURE__*/_react.default.createElement("rect", {
    x: 0,
    y: height / 2 - 0.25,
    width: width,
    height: height / 2 + 0.25,
    fill: secondaryColor
  }));
};
DiagLinesPattern.propTypes = {
  id: _propTypes.default.string.isRequired,
  color: _propTypes.default.string.isRequired,
  secondaryColor: _propTypes.default.string
};
DiagLinesPattern.defaultProps = {
  secondaryColor: '#ffffff'
};
const HalfAndHalfPattern = props => {
  const {
    id,
    color,
    secondaryColor
  } = props;
  return /*#__PURE__*/_react.default.createElement("pattern", {
    id: id,
    width: 1,
    height: 1,
    patternContentUnits: "objectBoundingBox"
  }, /*#__PURE__*/_react.default.createElement("path", {
    d: "M 0 0 V 1 L 1 0 Z",
    fill: color
  }), /*#__PURE__*/_react.default.createElement("path", {
    d: "M 1 1 V 0 L 0 1 Z",
    fill: secondaryColor
  }));
};
HalfAndHalfPattern.propTypes = {
  id: _propTypes.default.string.isRequired,
  color: _propTypes.default.string.isRequired,
  secondaryColor: _propTypes.default.string
};
HalfAndHalfPattern.defaultProps = {
  secondaryColor: '#ffffff'
};
const DiagHalfAndHalfPattern = props => {
  const {
    id,
    color,
    diagColor,
    diagColorFillOpacity,
    secondaryDiagColor
  } = props;
  const cellW = _AvailabilityUtils.SVG.CELL_WIDTH;
  const cellH = _AvailabilityUtils.SVG.CELL_HEIGHT;
  // Extends the bounds of the applied rectangle dimensions
  // when computing coordinates. This will overlay the
  // computed coordinate such that they extend beyond the actual bounds
  // of the rectangle, to account for squared stroke end of line points
  // so that there's no gaps with a sufficiently heavy stroke width.
  // Setting this to 0 will disable to the extension and map directly
  // onto the bounding box of the rectangle.
  // When divided by 2, should result in a rational number.
  const extendBoundsPadding = 0;
  const w = _AvailabilityUtils.SVG.CELL_WIDTH + extendBoundsPadding;
  const h = _AvailabilityUtils.SVG.CELL_HEIGHT + extendBoundsPadding;
  // Nudge the half fill color to prevent background from initial
  // line from top right corner consuming the color at normal
  // viewing levels. Ensures it looks visually appropriate for half color
  // fill and half diag line pattern.
  // Will be based on initDist
  const nudgeDiagFill = 1;
  const numLines = 4;
  // Distance of the diagonal of the rectangle
  const diagLengthRaw = Math.sqrt(cellW ** 2 + cellH ** 2);
  // Distance of initial line to top right corner, as opposed to starting
  // at point (0, w)
  const initDist = diagLengthRaw * 0.18;
  // Distance of furthest potential end line to bottom left corner
  const trailingDist = diagLengthRaw * 0.1;
  // Distance of the diagonal of the rectangle
  const diagLength = diagLengthRaw - (initDist + trailingDist);
  // Distance between parallel lines
  const diagLineGap = diagLength / numLines;
  const diagLineStrokeWidth = (diagLength * 0.1).toFixed(2);
  // Compute the initial x coordinate of the first line based on specified
  // initial distance from top right corner
  // 45 degree lines means perpendicular line to top right corner
  // c^2 = a^2 + b^2 where a == b
  // Find the initial x coordinate (y == 0) to derive linear function from
  const initXCoordDist = w - Math.sqrt(2 * initDist ** 2);
  const coords = [];
  // eslint-disable-next-line no-plusplus
  for (let i = 0; i < numLines; i++) {
    // Vertical transformation of linear function scalar value
    // As m == 1, use distance between parallel lines to compute hypotenuse
    // of equilateral triangle which will be the vertical transformation scalar
    // to apply to the linear function
    // Then scale by line index for each line
    const v = Math.sqrt(2 * diagLineGap ** 2) * i;
    // Linear functions derived from initial X coordinate, known slope
    const findY = x => x - initXCoordDist + v;
    const findX = y => y + initXCoordDist - v;
    let x1 = 0;
    let y1 = findY(x1);
    let x2 = w;
    let y2 = findY(x2);
    // Snap coordinates to bounding box
    if (y1 < 0) {
      x1 = findX(0);
      y1 = 0;
    }
    if (y2 > h) {
      x2 = findX(h);
      y2 = h;
    }
    coords.push({
      x1: (x1 - extendBoundsPadding / 2).toFixed(2),
      y1: (y1 - extendBoundsPadding / 2).toFixed(2),
      x2: (x2 - extendBoundsPadding / 2).toFixed(2),
      y2: (y2 - extendBoundsPadding / 2).toFixed(2)
    });
  }
  return /*#__PURE__*/_react.default.createElement("pattern", {
    id: id,
    x: 0,
    y: 0,
    width: 1,
    height: 1,
    patternUnits: "objectBoundingBox",
    patternContentUnits: "userSpaceOnUse"
  }, /*#__PURE__*/_react.default.createElement("polygon", {
    points: "0,0 0,1 1,1 1,0",
    fill: secondaryDiagColor,
    fillOpacity: diagColorFillOpacity
  }), coords.map((coord, idx) => /*#__PURE__*/_react.default.createElement("line", {
    // eslint-disable-next-line react/no-array-index-key
    key: "DiagHalfAndHalfPatternKey-".concat(coord.x1, "-").concat(coord.y1, "-").concat(coord.x2, "-").concat(coord.y2, "-").concat(idx),
    x1: coord.x1,
    y1: coord.y1,
    x2: coord.x2,
    y2: coord.y2,
    stroke: diagColor,
    strokeWidth: diagLineStrokeWidth,
    strokeLinecap: "square"
  })), /*#__PURE__*/_react.default.createElement("polygon", {
    points: "0,0 ".concat(cellW - nudgeDiagFill, ",0 ", 0, ",").concat(cellH),
    fill: color
  }));
};
DiagHalfAndHalfPattern.propTypes = {
  id: _propTypes.default.string.isRequired,
  color: _propTypes.default.string.isRequired,
  diagColor: _propTypes.default.string.isRequired,
  diagColorFillOpacity: _propTypes.default.number.isRequired,
  secondaryDiagColor: _propTypes.default.string
};
DiagHalfAndHalfPattern.defaultProps = {
  secondaryDiagColor: '#ffffff'
};
const SvgDefs = () => /*#__PURE__*/_react.default.createElement("svg", {
  width: "0px",
  height: "0px"
}, /*#__PURE__*/_react.default.createElement("defs", null, /*#__PURE__*/_react.default.createElement(DiagLinesPattern, {
  id: "availableProvisionalPattern",
  color: _Theme.COLORS.NEON_BLUE[700],
  secondaryColor: _Theme.COLORS.NEON_BLUE[50]
}), /*#__PURE__*/_react.default.createElement(DiagHalfAndHalfPattern, {
  id: "mixedAvailableProvisionalPattern",
  color: _Theme.COLORS.NEON_BLUE[700],
  diagColor: _Theme.COLORS.NEON_BLUE[700],
  diagColorFillOpacity: 0.25,
  secondaryDiagColor: _Theme.COLORS.NEON_BLUE[50]
}), /*#__PURE__*/_react.default.createElement(DiagLinesPattern, {
  id: "beingProcessedPattern",
  color: _Theme.COLORS.NEON_BLUE[700]
}), /*#__PURE__*/_react.default.createElement(DiagLinesPattern, {
  id: "delayedPattern",
  color: _Theme.COLORS.GOLD[400]
}), /*#__PURE__*/_react.default.createElement(DiagLinesPattern, {
  id: "partialSelectionPattern",
  color: _Theme.COLORS.LIGHT_BLUE[300],
  secondaryColor: _Theme.COLORS.LIGHT_BLUE[100]
}), /*#__PURE__*/_react.default.createElement(HalfAndHalfPattern, {
  id: "mixedSomeAvailabilityPattern",
  color: _Theme.COLORS.NEON_BLUE[700],
  secondaryColor: _Theme.COLORS.GOLD[400]
}), /*#__PURE__*/_react.default.createElement(HalfAndHalfPattern, {
  id: "mixedNoAvailabilityPattern",
  color: _Theme.default.palette.grey[200],
  secondaryColor: _Theme.COLORS.GOLD[400]
})));

/**
   CELLS
*/
exports.SvgDefs = SvgDefs;
const thinStrokeAttrs = {
  strokeWidth: '0.8px',
  width: "".concat(_AvailabilityUtils.SVG.CELL_WIDTH - 0.8, "px"),
  height: "".concat(_AvailabilityUtils.SVG.CELL_HEIGHT - 0.8, "px"),
  rx: "".concat(_AvailabilityUtils.SVG.CELL_RX * 1.5, "px"),
  nudge: 0.4
};
const midStrokeAttrs = {
  strokeWidth: '1.15px',
  width: "".concat(_AvailabilityUtils.SVG.CELL_WIDTH - 1.15, "px"),
  height: "".concat(_AvailabilityUtils.SVG.CELL_HEIGHT - 1.15, "px"),
  rx: "".concat(_AvailabilityUtils.SVG.CELL_RX * 1.25, "px"),
  nudge: 0.60
};
const fatStrokeAttrs = {
  strokeWidth: '1.5px',
  width: "".concat(_AvailabilityUtils.SVG.CELL_WIDTH - 1.5, "px"),
  height: "".concat(_AvailabilityUtils.SVG.CELL_HEIGHT - 1.5, "px"),
  rx: "".concat(_AvailabilityUtils.SVG.CELL_RX, "px"),
  nudge: 0.75
};
const noStrokeAttrs = {
  stroke: null,
  strokeWidth: null,
  width: "".concat(_AvailabilityUtils.SVG.CELL_WIDTH, "px"),
  height: "".concat(_AvailabilityUtils.SVG.CELL_HEIGHT, "px"),
  rx: "".concat(_AvailabilityUtils.SVG.CELL_RX * 2, "px"),
  nudge: 0
};
const CELL_ATTRS = exports.CELL_ATTRS = {
  available: _extends({
    fill: _Theme.COLORS.NEON_BLUE[700]
  }, noStrokeAttrs),
  'available-provisional': _extends({
    fill: 'url(#availableProvisionalPattern)',
    stroke: _Theme.COLORS.NEON_BLUE[700]
  }, midStrokeAttrs),
  'mixed-available-provisional': _extends({
    fill: 'url(#mixedAvailableProvisionalPattern)',
    stroke: _Theme.COLORS.NEON_BLUE[700]
  }, thinStrokeAttrs),
  'not available': _extends({
    fill: _Theme.default.palette.grey[200]
  }, noStrokeAttrs),
  tombstoned: _extends({
    // #708090 slate gray
    // #5A6673 darker slate gray
    // #727C8C more modified hue towards gray
    // #272727 greyscale of available blue color (rgba(39, 39, 39, 0.9))
    // Theme.palette.grey[500]
    fill: 'rgba(39, 39, 39, 0.9)'
  }, noStrokeAttrs),
  'not collected': _extends({
    fill: _Theme.COLORS.GOLD[400]
  }, noStrokeAttrs),
  expected: _extends({
    fill: '#ffffff',
    stroke: _Theme.COLORS.NEON_BLUE[700]
  }, fatStrokeAttrs),
  tentative: _extends({
    fill: '#ffffff',
    stroke: _Theme.COLORS.BROWN[300]
  }, fatStrokeAttrs),
  'not expected': _extends({
    fill: '#ffffff',
    stroke: _Theme.default.palette.grey[200]
  }, fatStrokeAttrs),
  'being processed': _extends({
    fill: 'url(#beingProcessedPattern)',
    stroke: _Theme.COLORS.NEON_BLUE[700]
  }, fatStrokeAttrs),
  delayed: _extends({
    fill: 'url(#delayedPattern)',
    stroke: _Theme.COLORS.GOLD[400]
  }, fatStrokeAttrs),
  'mixed some availability': _extends({
    fill: 'url(#mixedSomeAvailabilityPattern)',
    stroke: _Theme.COLORS.NEON_BLUE[700]
  }, thinStrokeAttrs),
  'mixed no availability': _extends({
    fill: 'url(#mixedNoAvailabilityPattern)',
    stroke: _Theme.COLORS.GOLD[400]
  }, thinStrokeAttrs)
};
const JsxCell = props => {
  const {
    status,
    x,
    y
  } = props;
  const _CELL_ATTRS$status = CELL_ATTRS[status],
    {
      nudge = 0
    } = _CELL_ATTRS$status,
    attrs = _objectWithoutPropertiesLoose(_CELL_ATTRS$status, _excluded);
  return /*#__PURE__*/_react.default.createElement("rect", _extends({
    x: x + nudge,
    y: y + nudge,
    rx: "".concat(_AvailabilityUtils.SVG.CELL_RX, "px")
  }, attrs));
};
exports.JsxCell = JsxCell;
JsxCell.propTypes = {
  status: _propTypes.default.oneOf(Object.keys(_AvailabilityUtils.VALID_ENHANCED_STATUSES)).isRequired,
  x: _propTypes.default.number,
  y: _propTypes.default.number
};
JsxCell.defaultProps = {
  x: 0,
  y: 0
};