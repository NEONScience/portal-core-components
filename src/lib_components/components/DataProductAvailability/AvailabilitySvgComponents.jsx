/* eslint-disable import/prefer-default-export, max-len */

import React from 'react';
import PropTypes from 'prop-types';

// import { select } from 'd3-selection';

import { SVG, VALID_ENHANCED_STATUSES } from './AvailabilityUtils';

import Theme, { COLORS } from '../Theme/Theme';

/**
   SVG_DEFS
*/
const DiagLinesPattern = (props) => {
  const { id, color, secondaryColor } = props;
  const width = SVG.CELL_WIDTH;
  const height = SVG.CELL_HEIGHT / 4;
  return (
    <pattern
      id={id}
      width={width}
      height={height}
      patternUnits="userSpaceOnUse"
      patternTransform="rotate(45)"
    >
      <rect x={0} y={0} width={width} height={(height / 2) - 0.25} fill={color} />
      <rect x={0} y={(height / 2) - 0.25} width={width} height={(height / 2) + 0.25} fill={secondaryColor} />
    </pattern>
  );
};
DiagLinesPattern.propTypes = {
  id: PropTypes.string.isRequired,
  color: PropTypes.string.isRequired,
  secondaryColor: PropTypes.string,
};
DiagLinesPattern.defaultProps = {
  secondaryColor: '#ffffff',
};
const HalfAndHalfPattern = (props) => {
  const { id, color, secondaryColor } = props;
  return (
    <pattern
      id={id}
      width={1}
      height={1}
      patternContentUnits="objectBoundingBox"
    >
      <path d="M 0 0 V 1 L 1 0 Z" fill={color} />
      <path d="M 1 1 V 0 L 0 1 Z" fill={secondaryColor} />
    </pattern>
  );
};
HalfAndHalfPattern.propTypes = {
  id: PropTypes.string.isRequired,
  color: PropTypes.string.isRequired,
  secondaryColor: PropTypes.string,
};
HalfAndHalfPattern.defaultProps = {
  secondaryColor: '#ffffff',
};

export const SvgDefs = () => (
  <svg width="0px" height="0px">
    <defs>
      <DiagLinesPattern id="availableProvisionalPattern" color={COLORS.NEON_BLUE[700]} secondaryColor={COLORS.NEON_BLUE[100]} />
      <DiagLinesPattern id="beingProcessedPattern" color={COLORS.NEON_BLUE[700]} />
      <DiagLinesPattern id="delayedPattern" color={COLORS.GOLD[400]} />
      <DiagLinesPattern id="partialSelectionPattern" color={COLORS.LIGHT_BLUE[300]} secondaryColor={COLORS.LIGHT_BLUE[100]} />
      <HalfAndHalfPattern id="mixedSomeAvailabilityPattern" color={COLORS.NEON_BLUE[700]} secondaryColor={COLORS.GOLD[400]} />
      <HalfAndHalfPattern id="mixedNoAvailabilityPattern" color={Theme.palette.grey[200]} secondaryColor={COLORS.GOLD[400]} />
    </defs>
  </svg>
);

/**
   CELLS
*/
const thinStrokeAttrs = {
  strokeWidth: '0.8px',
  width: `${SVG.CELL_WIDTH - 0.8}px`,
  height: `${SVG.CELL_HEIGHT - 0.8}px`,
  rx: `${SVG.CELL_RX * 1.5}px`,
  nudge: 0.4,
};
const fatStrokeAttrs = {
  strokeWidth: '1.5px',
  width: `${SVG.CELL_WIDTH - 1.5}px`,
  height: `${SVG.CELL_HEIGHT - 1.5}px`,
  rx: `${SVG.CELL_RX}px`,
  nudge: 0.75,
};
const noStrokeAttrs = {
  stroke: null,
  strokeWidth: null,
  width: `${SVG.CELL_WIDTH}px`,
  height: `${SVG.CELL_HEIGHT}px`,
  rx: `${SVG.CELL_RX * 2}px`,
  nudge: 0,
};
export const CELL_ATTRS = {
  available: {
    fill: COLORS.NEON_BLUE[700],
    ...noStrokeAttrs,
  },
  'available-provisional': {
    fill: 'url(#availableProvisionalPattern)',
    stroke: COLORS.NEON_BLUE[700],
    ...fatStrokeAttrs,
  },
  'not available': {
    fill: Theme.palette.grey[200],
    ...noStrokeAttrs,
  },
  'not collected': {
    fill: COLORS.GOLD[400],
    ...noStrokeAttrs,
  },
  expected: {
    fill: '#ffffff',
    stroke: COLORS.NEON_BLUE[700],
    ...fatStrokeAttrs,
  },
  tentative: {
    fill: '#ffffff',
    stroke: COLORS.BROWN[300],
    ...fatStrokeAttrs,
  },
  'not expected': {
    fill: '#ffffff',
    stroke: Theme.palette.grey[200],
    ...fatStrokeAttrs,
  },
  'being processed': {
    fill: 'url(#beingProcessedPattern)',
    stroke: COLORS.NEON_BLUE[700],
    ...fatStrokeAttrs,
  },
  delayed: {
    fill: 'url(#delayedPattern)',
    stroke: COLORS.GOLD[400],
    ...fatStrokeAttrs,
  },
  'mixed some availability': {
    fill: 'url(#mixedSomeAvailabilityPattern)',
    stroke: COLORS.NEON_BLUE[700],
    ...thinStrokeAttrs,
  },
  'mixed no availability': {
    fill: 'url(#mixedNoAvailabilityPattern)',
    stroke: COLORS.GOLD[400],
    ...thinStrokeAttrs,
  },
};

export const JsxCell = (props) => {
  const { status, x, y } = props;
  const { nudge = 0, ...attrs } = CELL_ATTRS[status];
  return (
    <rect x={x + nudge} y={y + nudge} rx={`${SVG.CELL_RX}px`} {...attrs} />
  );
};
JsxCell.propTypes = {
  status: PropTypes.oneOf(Object.keys(VALID_ENHANCED_STATUSES)).isRequired,
  x: PropTypes.number,
  y: PropTypes.number,
};
JsxCell.defaultProps = { x: 0, y: 0 };
