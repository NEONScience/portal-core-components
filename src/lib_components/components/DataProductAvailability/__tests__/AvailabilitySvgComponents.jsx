import React from 'react';
import renderer from 'react-test-renderer';

import { SvgDefs, JsxCell, CELL_ATTRS } from '../AvailabilitySvgComponents';

describe('DataProductAvailability - AvailabilitySvgComponents', () => {
  test('renders SvgDefs correctly', () => {
    const tree = renderer
      .create(<SvgDefs />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  describe('renders JsxCell correctly for every status', () => {
    Object.keys(CELL_ATTRS).forEach((status) => {
      test(`JsxCell for status: ${status}`, () => {
        const tree = renderer
          .create(<JsxCell status={status} x={10} y={20} />)
          .toJSON();
        expect(tree).toMatchSnapshot();
      });
    });
  });
});
