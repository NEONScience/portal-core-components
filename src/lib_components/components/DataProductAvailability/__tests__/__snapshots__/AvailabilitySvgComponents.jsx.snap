// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataProductAvailability - AvailabilitySvgComponents renders JsxCell correctly for every status JsxCell for status: available 1`] = `
<rect
  fill="#002c77"
  height="12px"
  rx="2px"
  stroke={null}
  strokeWidth={null}
  width="7px"
  x={10}
  y={20}
/>
`;

exports[`DataProductAvailability - AvailabilitySvgComponents renders JsxCell correctly for every status JsxCell for status: available-provisional 1`] = `
<rect
  fill="url(#availableProvisionalPattern)"
  height="10.85px"
  rx="1.25px"
  stroke="#002c77"
  strokeWidth="1.15px"
  width="5.85px"
  x={10.6}
  y={20.6}
/>
`;

exports[`DataProductAvailability - AvailabilitySvgComponents renders JsxCell correctly for every status JsxCell for status: being processed 1`] = `
<rect
  fill="url(#beingProcessedPattern)"
  height="10.5px"
  rx="1px"
  stroke="#002c77"
  strokeWidth="1.5px"
  width="5.5px"
  x={10.75}
  y={20.75}
/>
`;

exports[`DataProductAvailability - AvailabilitySvgComponents renders JsxCell correctly for every status JsxCell for status: delayed 1`] = `
<rect
  fill="url(#delayedPattern)"
  height="10.5px"
  rx="1px"
  stroke="#f8bb32"
  strokeWidth="1.5px"
  width="5.5px"
  x={10.75}
  y={20.75}
/>
`;

exports[`DataProductAvailability - AvailabilitySvgComponents renders JsxCell correctly for every status JsxCell for status: expected 1`] = `
<rect
  fill="#ffffff"
  height="10.5px"
  rx="1px"
  stroke="#002c77"
  strokeWidth="1.5px"
  width="5.5px"
  x={10.75}
  y={20.75}
/>
`;

exports[`DataProductAvailability - AvailabilitySvgComponents renders JsxCell correctly for every status JsxCell for status: mixed no availability 1`] = `
<rect
  fill="url(#mixedNoAvailabilityPattern)"
  height="11.2px"
  rx="1.5px"
  stroke="#f8bb32"
  strokeWidth="0.8px"
  width="6.2px"
  x={10.4}
  y={20.4}
/>
`;

exports[`DataProductAvailability - AvailabilitySvgComponents renders JsxCell correctly for every status JsxCell for status: mixed some availability 1`] = `
<rect
  fill="url(#mixedSomeAvailabilityPattern)"
  height="11.2px"
  rx="1.5px"
  stroke="#002c77"
  strokeWidth="0.8px"
  width="6.2px"
  x={10.4}
  y={20.4}
/>
`;

exports[`DataProductAvailability - AvailabilitySvgComponents renders JsxCell correctly for every status JsxCell for status: mixed-available-provisional 1`] = `
<rect
  fill="url(#mixedAvailableProvisionalPattern)"
  height="11.2px"
  rx="1.5px"
  stroke="#002c77"
  strokeWidth="0.8px"
  width="6.2px"
  x={10.4}
  y={20.4}
/>
`;

exports[`DataProductAvailability - AvailabilitySvgComponents renders JsxCell correctly for every status JsxCell for status: not available 1`] = `
<rect
  fill="#d7d9d9"
  height="12px"
  rx="2px"
  stroke={null}
  strokeWidth={null}
  width="7px"
  x={10}
  y={20}
/>
`;

exports[`DataProductAvailability - AvailabilitySvgComponents renders JsxCell correctly for every status JsxCell for status: not collected 1`] = `
<rect
  fill="#f8bb32"
  height="12px"
  rx="2px"
  stroke={null}
  strokeWidth={null}
  width="7px"
  x={10}
  y={20}
/>
`;

exports[`DataProductAvailability - AvailabilitySvgComponents renders JsxCell correctly for every status JsxCell for status: not expected 1`] = `
<rect
  fill="#ffffff"
  height="10.5px"
  rx="1px"
  stroke="#d7d9d9"
  strokeWidth="1.5px"
  width="5.5px"
  x={10.75}
  y={20.75}
/>
`;

exports[`DataProductAvailability - AvailabilitySvgComponents renders JsxCell correctly for every status JsxCell for status: tentative 1`] = `
<rect
  fill="#ffffff"
  height="10.5px"
  rx="1px"
  stroke="#b39076"
  strokeWidth="1.5px"
  width="5.5px"
  x={10.75}
  y={20.75}
/>
`;

exports[`DataProductAvailability - AvailabilitySvgComponents renders JsxCell correctly for every status JsxCell for status: tombstoned 1`] = `
<rect
  fill="rgba(39, 39, 39, 0.9)"
  height="12px"
  rx="2px"
  stroke={null}
  strokeWidth={null}
  width="7px"
  x={10}
  y={20}
/>
`;

exports[`DataProductAvailability - AvailabilitySvgComponents renders SvgDefs correctly 1`] = `
<svg
  height="0px"
  width="0px"
>
  <defs>
    <pattern
      height={3}
      id="availableProvisionalPattern"
      patternTransform="rotate(45)"
      patternUnits="userSpaceOnUse"
      width={7}
    >
      <rect
        fill="#002c77"
        height={1.25}
        width={7}
        x={0}
        y={0}
      />
      <rect
        fill="#e1e3ea"
        height={1.75}
        width={7}
        x={0}
        y={1.25}
      />
    </pattern>
    <pattern
      height={1}
      id="mixedAvailableProvisionalPattern"
      patternContentUnits="userSpaceOnUse"
      patternUnits="objectBoundingBox"
      width={1}
      x={0}
      y={0}
    >
      <polygon
        fill="#e1e3ea"
        fillOpacity={0.25}
        points="0,0 0,1 1,1 1,0"
      />
      <line
        stroke="#002c77"
        strokeLinecap="square"
        strokeWidth="1.00"
        x1="3.46"
        x2="7.00"
        y1="0.00"
        y2="3.54"
      />
      <line
        stroke="#002c77"
        strokeLinecap="square"
        strokeWidth="1.00"
        x1="0.00"
        x2="7.00"
        y1="0.07"
        y2="7.07"
      />
      <line
        stroke="#002c77"
        strokeLinecap="square"
        strokeWidth="1.00"
        x1="0.00"
        x2="7.00"
        y1="3.61"
        y2="10.61"
      />
      <line
        stroke="#002c77"
        strokeLinecap="square"
        strokeWidth="1.00"
        x1="0.00"
        x2="4.85"
        y1="7.15"
        y2="12.00"
      />
      <polygon
        fill="#002c77"
        points="0,0 6,0 0,12"
      />
    </pattern>
    <pattern
      height={3}
      id="beingProcessedPattern"
      patternTransform="rotate(45)"
      patternUnits="userSpaceOnUse"
      width={7}
    >
      <rect
        fill="#002c77"
        height={1.25}
        width={7}
        x={0}
        y={0}
      />
      <rect
        fill="#ffffff"
        height={1.75}
        width={7}
        x={0}
        y={1.25}
      />
    </pattern>
    <pattern
      height={3}
      id="delayedPattern"
      patternTransform="rotate(45)"
      patternUnits="userSpaceOnUse"
      width={7}
    >
      <rect
        fill="#f8bb32"
        height={1.25}
        width={7}
        x={0}
        y={0}
      />
      <rect
        fill="#ffffff"
        height={1.75}
        width={7}
        x={0}
        y={1.25}
      />
    </pattern>
    <pattern
      height={3}
      id="partialSelectionPattern"
      patternTransform="rotate(45)"
      patternUnits="userSpaceOnUse"
      width={7}
    >
      <rect
        fill="#5ca6e3"
        height={1.25}
        width={7}
        x={0}
        y={0}
      />
      <rect
        fill="#b8d8f4"
        height={1.75}
        width={7}
        x={0}
        y={1.25}
      />
    </pattern>
    <pattern
      height={1}
      id="mixedSomeAvailabilityPattern"
      patternContentUnits="objectBoundingBox"
      width={1}
    >
      <path
        d="M 0 0 V 1 L 1 0 Z"
        fill="#002c77"
      />
      <path
        d="M 1 1 V 0 L 0 1 Z"
        fill="#f8bb32"
      />
    </pattern>
    <pattern
      height={1}
      id="mixedNoAvailabilityPattern"
      patternContentUnits="objectBoundingBox"
      width={1}
    >
      <path
        d="M 0 0 V 1 L 1 0 Z"
        fill="#d7d9d9"
      />
      <path
        d="M 1 1 V 0 L 0 1 Z"
        fill="#f8bb32"
      />
    </pattern>
  </defs>
</svg>
`;
